packages/sirius-web/frontend/sirius-web-application/src/modals/duplicate-object/DuplicateObjectModal.tsx

```diff
@@ -0,0 +1,270 @@
+import DialogTitle from '@mui/material/DialogTitle';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import IconButton from '@mui/material/IconButton';
+import MenuItem from '@mui/material/MenuItem';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { useEffect, useState } from 'react';
+import { makeStyles } from 'tss-react/mui';
+import { DuplicateObjectModalProps, DuplicateObjectModalState } from './DuplicateObjectModal.types';
+import { FilterBar } from './FilterBar';
+import { useContainmentFeatureNames } from './useContainmentFeatureNames';
+import { useDuplicateObject } from './useDuplicateObject';
+import { useDuplicateDialogTreeSubscription } from './useDuplicateDialogTreeSubscription';
+
+const useDuplicateObjectModalStyles = makeStyles()((theme) => ({
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    '& > *': {
+      marginBottom: theme.spacing(1),
+    },
+  },
+  title: {
+    opacity: 0.6,
+    fontSize: theme.typography.caption.fontSize,
+  },
+  borderStyle: {
+    border: '1px solid',
+    borderColor: theme.palette.grey[500],
+    height: 300,
+    overflow: 'auto',
+  },
+}));
+
+export const DuplicateObjectModal = ({
+  editingContextId,
+  objectToDuplicateId,
+  objectToDuplicateKind,
+  onObjectDuplicated,
+  onClose,
+}: DuplicateObjectModalProps) => {
+  const { classes } = useDuplicateObjectModalStyles();
+
+  const [state, setState] = useState<DuplicateObjectModalState>({
+    containerSelection: { entries: [] },
+    containmentFeatureNames: [],
+    containmentFeatureName: '',
+    duplicateContent: true,
+    copyOutgoingReferences: true,
+    updateIncomingReferences: false,
+    filterBarText: '',
+    expanded: [],
+    maxDepth: 1,
+  });
+
+  const { getContainmentFeatureNames, containmentFeatureNames } = useContainmentFeatureNames();
+
+  useEffect(() => {
+    if (state.containerSelection.entries[0]) {
+      getContainmentFeatureNames({
+        variables: {
+          editingContextId,
+          containerId: state.containerSelection.entries[0].id,
+          containedObjectId: objectToDuplicateId,
+        },
+      });
+    }
+  }, [state.containerSelection, objectToDuplicateId]);
+
+  useEffect(() => {
+    if (containmentFeatureNames) {
+      setState((prevState) => ({
+        ...prevState,
+        containmentFeatureNames,
+        containmentFeatureName: containmentFeatureNames[0] ?? '',
+      }));
+    }
+  }, [containmentFeatureNames]);
+
+  const { duplicateObject, duplicatedObject } = useDuplicateObject();
+  const onDuplicate = () => {
+    duplicateObject(
+      editingContextId,
+      objectToDuplicateId,
+      state.containerSelection.entries[0]?.id,
+      state.containmentFeatureName,
+      state.duplicateContent,
+      state.copyOutgoingReferences,
+      state.updateIncomingReferences
+    );
+  };
+  useEffect(() => {
+    if (duplicatedObject) {
+      onObjectDuplicated({ entries: [duplicatedObject] });
```

Comment:
DuplicatedObject has more properties than what is expected in the selection entry object. Please just copy what's necessary